web3.js/test/transaction.test.ts

```diff
@@ -962,4 +963,48 @@ describe('Transaction', () => {
     t1.partialSign(signer);
     t1.serialize();
   });
+
+  it('externally signed versioned transaction', () => {
+    const signer = Keypair.generate();
+    const invalidSigner = Keypair.generate();
+
+    const recentBlockhash = new PublicKey(3).toBuffer();
+
+    const message = new TransactionMessage({
+      payerKey: signer.publicKey,
+      instructions: [
+        new TransactionInstruction({
+          data: Buffer.from('Hello!'),
+          keys: [
+            {
+              pubkey: signer.publicKey,
+              isSigner: true,
+              isWritable: true,
+            },
+          ],
+          programId: new PublicKey(
+            'MemoSq4gqABAXKb96qnH8TysNcWxMyWCqXgDLGmfcHr',
+          ),
+        }),
+      ],
+      recentBlockhash: bs58.encode(recentBlockhash),
+    });
+
+    const transaction = new VersionedTransaction(message.compileToV0Message());
+
+    const signature = sign(transaction.message.serialize(), signer.secretKey);
+
+    transaction.addSignature(signer.publicKey, signature);
+
+    expect(transaction.signatures).to.have.length(1);
+    expect(transaction.signatures[0]).to.eq(signature);
```

Comment:
Missing from this is a test that exercises that the signature is appended at the right index. Create a transaction with two signers and assert that the right signatures end up at the right indices in the array.